src/storage/index.ts

```diff
@@ -784,12 +791,15 @@ export async function putFile(
     const signatureContent = JSON.stringify(signatureObject)
 
     uploadFn = async (hubConfig: GaiaHubConfig) => {
-      const fileUrls = await Promise.all([
-        uploadToGaiaHub(path, contentData, hubConfig, contentType),
+      const writeResponse = (await Promise.all([
+        uploadToGaiaHub(path, contentData, hubConfig, contentType, etags[path]),
```

Comment:
The concurrency situation with `signed & unencrypted` uploads here is pretty bad. We should create an issue for attempting to reduce this to a single file that contains both the signature and data (similar to the `signed & encrypted` payload).

Reply:
done